bin/server.dart

```diff
@@ -108,3 +129,29 @@ extension HttpResponseExtensions on HttpResponse {
 }
 
 bool get isProduction => Platform.environment['ENVIRONMENT'] == 'prod';
+
+extension IsolateExtensions on Isolate {
+  void addSentryErrorListener() {
+    final receivePort = RawReceivePort((dynamic values) async {
+      await SentryIsolateIntegration.captureIsolateError(values);
+    });
+
+    Isolate.current.addErrorListener(receivePort.sendPort);
+  }
+}
+
+class SentryIsolateIntegration {
+  static Future<void> captureIsolateError(dynamic error) {
+    if (error is List<dynamic> && error.length == 2) {
+      /// https://api.dart.dev/stable/2.9.0/dart-isolate/Isolate/addErrorListener.html
+      dynamic stackTrace = error[1];
+      if (stackTrace != null) {
+        stackTrace = StackTrace.fromString(stackTrace as String);
+      }
+      return Sentry.captureException(error[0], stackTrace: stackTrace);
```

Comment:
the `stackTrace` is a dynamic param, pass as it is, do not convert to a String, it might be a `StackTrace` or even null, the `encodeStackTrace` method already checks its type and do the right thing, so I suggest just removing:

```
if (stackTrace != null) {
        stackTrace = StackTrace.fromString(stackTrace as String);
      }
```

Reply:
Also for Isolates? This code came from somewhere and I can't recall details of it from the top of my head